crates/trie/sparse/src/trie.rs

```diff
@@ -31,13 +45,36 @@ impl TrieMasks {
     }
 }
 
-/// Inner representation of the sparse trie.
-/// Sparse trie is blind by default until nodes are revealed.
-#[derive(PartialEq, Eq)]
+/// A sparse trie that is either in a "blind" state (nodes are hidden and only represented
+/// by their hashes) or in a "revealed" state (nodes have been loaded and can be updated).
+///
+/// In blind mode the trie does not contain any decoded node data, which saves memory but
+/// prevents direct access to node contents. The revealed mode stores decoded nodes along
+/// with additional information such as values, allowing direct manipulation.
+///
+/// The sparse trie design is optimised for:
+/// 1. Memory efficiency - only needed nodes are loaded into memory
+/// 2. Update tracking - changes to the trie structure can be tracked and selectively persisted
+/// 3. Incremental operations - nodes can be revealed as needed without loading the entire trie
+///
+/// One way to visualize blind vs revealed: Imagine we take the Ethereum Trie that represents
+/// all state, and we started to "hide" some of the nodes. If we looked for that node in the
+/// Trie, it would not be available, but it does not mean that it is not there. Importantly,
+/// a "hidden" or "blind" node does not mean that it does not exists, it just means that it has
+/// not been revealed.
```

Comment:
I don't think this description is accurate, `Blind` here means that we did not reveal any nodes in the sparse trie, but once we reveal at least the root node, it turns into `Revealed`

Reply:
Makes sense -- I was talking about particular nodes/subtrees being blinded vs the whole sparse trie

Reply:
Removed that paragraph for simplicity